test/e2e/external_gateways.go

```diff
@@ -1934,6 +1934,325 @@ var _ = ginkgo.Describe("External Gateway test suite", func() {
 			})
 		})
 	})
+
+	var _ = ginkgo.Context("When migrating from Annotations to Admin Policy Based External Route CRs", func() {
+		// Validate pods can reach a network running in a container's looback address via
+		// an external gateway running on eth0 of the container without any tunnel encap.
+		// The traffic will get proxied through an annotated pod in the serving namespace.
+		var _ = ginkgo.Describe("e2e non-vxlan external gateway through a gateway pod", func() {
+			const (
+				svcname         string = "externalgw-pod-novxlan"
+				gwContainer1    string = "ex-gw-container1"
+				gwContainer2    string = "ex-gw-container2"
+				srcPingPodName  string = "e2e-exgw-src-ping-pod"
+				gatewayPodName1 string = "e2e-gateway-pod1"
+				gatewayPodName2 string = "e2e-gateway-pod2"
+				externalTCPPort        = 91
+				externalUDPPort        = 90
+				ecmpRetry       int    = 20
+				testTimeout     string = "20"
+			)
+
+			var (
+				sleepCommand             = []string{"bash", "-c", "sleep 20000"}
+				addressesv4, addressesv6 gatewayTestIPs
+				clientSet                kubernetes.Interface
+				servingNamespace         string
+			)
+
+			var (
+				gwContainers []string
+			)
+
+			f := wrappedTestFramework(svcname)
+
+			ginkgo.BeforeEach(func() {
+				clientSet = f.ClientSet // so it can be used in AfterEach
+				// retrieve worker node names
+				nodes, err := e2enode.GetBoundedReadySchedulableNodes(f.ClientSet, 3)
+				framework.ExpectNoError(err)
+				if len(nodes.Items) < 3 {
+					framework.Failf(
+						"Test requires >= 3 Ready nodes, but there are only %v nodes",
+						len(nodes.Items))
+				}
+
+				ns, err := f.CreateNamespace("exgw-serving", nil)
+				framework.ExpectNoError(err)
+				servingNamespace = ns.Name
+
+				gwContainers, addressesv4, addressesv6 = setupGatewayContainers(f, nodes, gwContainer1, gwContainer2, srcPingPodName, externalUDPPort, externalTCPPort, ecmpRetry)
+				setupAnnotatedGatewayPods(f, nodes, gatewayPodName1, gatewayPodName2, servingNamespace, sleepCommand, addressesv4, addressesv6, false)
```

Comment:
so before each test you setup gw annotations. Then you create the policy CRD, remove the annotations and verify the outcome right?

Reply:
no thats fine. I just wanted to make sure I understood it correctly.